### PR TITLE
Addition of Jet variables/gen event histograms to the susy multilepton analysis/heppy updates (jet-tau, LHE analyzer and PDF weights)

### PR DESCRIPTION
**Pull request by mmarionncern**
Add
- 30 GeV b jet counter
- extra jet variables for the susy multilepton analysis
- supercluster eta value for electrons
  -switch for the output mode of the susy multilepton analysis + gen event counter histogram 
- counters of number of jets not matched with taus
- list of indices for jets not matched with taus
- heppy modification on jet-tau/LHE analyzer and PDF weights, as specified in #287

### CODE REVIEW

**Comment by cbotta** [comment]
@mmarionncern The 30GeV counters for jets (both b-jets and all-jets) are needed to reproduce the ra7 baseline analysis, for which, as discussed today we need to clean the jets from the taus. So I would first prepare the tau-jet cleaning as discussed privately, and once we have the new collections/flags we can add those counters as well

File: CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyMultilepton.py [hunk 1]
```diff
@@ -126,4 +127,9 @@
             "ivf"       : NTupleCollection("SV",     svType, 20, help="SVs from IVF"),
             "genBHadrons"  : NTupleCollection("GenBHad", heavyFlavourHadronType, 20, mcOnly=True, help="Gen-level B hadrons"),
             "genDHadrons"  : NTupleCollection("GenDHad", heavyFlavourHadronType, 20, mcOnly=True, help="Gen-level D hadrons"),
+            ##------------------------------------------------
+            "jetsNonTauIdx" : NTupleCollection("JetNoTauIdx",objectInt, 10, help="index of jets not associated to taus"),
+            "bjetsMNonTauIdx" : NTupleCollection("BJetMediumNoTauIdx",objectInt, 10, help="index of b-jets (CSV medium) not associated to taus"),
```
Comment (gpetruc) [inline CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyMultilepton.py]: Why do you need `bjetsMNonTauIdx` ? we're not saving the colleciton of b-jets so this is probably unnecessary.

File: CMGTools/TTHAnalysis/python/analyzers/ttHJetTauAnalyzer.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        for ij in xrange(len(jets)):
+            if not jets[ij].taus:
+                iJetNoTau.append(ij)
+                
+        return iJetNoTau
+
+    def process(self, event):
+        self.readCollections(event.input)
+        
+        for j in event.cleanjets
```
Comment (gpetruc) [inline CMGTools/TTHAnalysis/python/analyzers/ttHJetTauAnalyzer.py]: this line doesn't even compile; leftover of some cleanup?

Comment (mmarionncern) [inline CMGTools/TTHAnalysis/python/analyzers/ttHJetTauAnalyzer.py]: leftover of debugging, removed

File: CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyMultilepton.py (hunk 1 above)
Comment (mmarionncern) [inline CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyMultilepton.py]: I forgot that last point when adding it, removed

File: CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyCore.py [hunk 3]
```diff
@@ -4,16 +4,28 @@
...
+            
+            NTupleVariable("nJet30", lambda ev: sum([j.pt() > 30 for j in ev.cleanJets]), int, help="Number of jets with pt > 30, |eta|<2.4"),
+            NTupleVariable("nJet30a", lambda ev: sum([j.pt() > 30 for j in ev.cleanJetsAll]), int, help="Number of jets with pt > 30, |eta|<4.7"),
+            NTupleVariable("nJet30NoTau", lambda ev: sum([ (j.pt() > 30 and not j.taus ) for j in ev.cleanJets]), int, help="Number of jets with pt > 30, not matched with taus"),
+            NTupleVariable("nBJetLoose30", lambda ev: sum([j.btagWP("CSVv2IVFL") for j in ev.cleanJets if j.pt() > 30]), int, help="Number of jets with pt > 30 passing CSV loose"),
+            NTupleVariable("nBJetMedium30", lambda ev: sum([j.btagWP("CSVv2IVFM") for j in ev.bjetsMedium if j.pt() > 30]), int, help="Number of jets with pt > 30 passing CSV medium"),
+            NTupleVariable("nBJetMedium30NoTau", lambda ev: sum([(j.btagWP("CSVv2IVFM") and not j.taus) for j in ev.bjetsMedium if j.pt() > 30]), int, help="Number of jets with pt > 30 passing CSV medium, not matched with taus"),
```
Comment (cbotta) [inline CMGTools/TTHAnalysis/python/analyzers/treeProducerSusyCore.py:23]: can you also add nBJetLoose25/30/40NoTau? (in the last ra7 meeting people were talking about using also the Loose b-tagged jets for some SRs) Thanks.

**Comment by mmarionncern** [comment]
Counters for loose csv bjets non matched with taus added

**Pull request merged**